ci/jenkins/stop-stale-jobs.sh

```diff
@@ -0,0 +1,171 @@
+            jenkins_variable="$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json")"
+            pr_id=$(echo "$jenkins_variable" | jq -r "[.[]][1].ghprbPullId")
+
+            if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+                echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+                # Need to stop first then clean.
+                curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN || true
+                
+                # Ensures that the job has been aborted before cleanup.
+                retry=5
+                flag=0
+                while [[ "${retry}" -gt 0 ]]; do
+                    status=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/$last_build_number/api/json" | jq -r "[.[]][14]")
+                    if [[ "$status" == "ABORTED" ]]; then
+                         flag=1
+                         break
+                    fi
+                    sleep 5
+                    retry=$((retry-1)) 
+                done
+                if [[ $flag == 1 ]]; then
+
+                    CLUSTER_NAME=$(echo "$jenkins_variable" | jq -r "[.[]][1].BUILD_TAG")
+                    ip_of_capv_testbed=$(echo "$jenkins_variable" | jq "[.[]][1].SSH_CONNECTION" | awk '{print $3}')
+
+                    cleanup_running_job "$CLUSTER_NAME" "$ip_of_capv_testbed"
+                else
+                    echo "Job $capv_job_name still running on PR ${PULL_REQUEST_ID}"
+                fi
+            fi
+        fi   
+    done
+}
+
+function stop_all() {
+     # Check all stale waiting jobs on PR.
+      abort_queue_jobs
+
+     # Check all stale running jobs on PR.
+      abort_running_jobs   
+}
+
+# Ensures that the script can be run from anywhere.
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+GIT_CHECKOUT_DIR=${THIS_DIR}/../..
+pushd "$THIS_DIR" > /dev/null
+
+function release_static_ip {
+    CLUSTER_NAME=$1
+    ip_of_capv_testbed=$2
+    echo '=== Releasing IP ==='
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} "cat "$DEFAULT_WORKDIR/host-local.json" | \
+    CNI_COMMAND=DEL CNI_CONTAINERID="$CLUSTER_NAME" \
+    CNI_NETNS=/dev/null CNI_IFNAME=dummy0 CNI_PATH=. /usr/bin/host-local"
+    echo "Released IP"
+}
+
+function cleanup_running_job {
+    CLUSTER_NAME=$1
+    ip_of_capv_testbed=$2
+    echo "=== Cleaning up VMC cluster ${CLUSTER_NAME} ==="
+    # To run kubectl cmds.
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} kubectl delete cluster ${CLUSTER_NAME} -n ${CLUSTER_NAME}
+    ssh -o StrictHostKeyChecking=no -i /var/lib/jenkins/.ssh/id_rsa -n jenkins@${ip_of_capv_testbed} kubectl delete ns ${CLUSTER_NAME}
+    rm -rf "${GIT_CHECKOUT_DIR}/jenkins"
```

Comment:
May I know why we need to delete this directory and how is it created?

Reply:
Actually initially I thought maybe we need removed this dir also to perform cleanup because it has been used in capv testbed but I think  it's not required. thanks!